src/System.Runtime.Extensions/src/System/Environment.WinRT.cs

```diff
@@ -0,0 +1,124 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Windows.Foundation.Metadata;
+using Windows.Storage;
+
+namespace System
+{
+    public static partial class Environment
+    {
+        private static string GetFolderPathCore(SpecialFolder folder, SpecialFolderOption option)
+        {
+            // For testing we'll fall back if the needed APIs aren't present
+            if (ApiInformation.IsTypePresent("Windows.Storage.UserDataPaths"))
+            {
+                return GetFolderPathCoreCurrent(folder, option);
+            }
+            else
+            {
+                return GetFolderPathCoreFallBack(folder, option);
+            }
+
+        }
+
+        private static string GetFolderPathCoreCurrent(SpecialFolder folder, SpecialFolderOption option)
```

Comment:
Is it correct that we're ignoring `option`, both here and in the fallback?  We implement that manually in the Unix implementation, for example:
https://github.com/dotnet/corefx/blob/475c578fc77e16344679e337276c91a70fe1c79c/src/System.Runtime.Extensions/src/System/Environment.Unix.cs#L67
https://github.com/dotnet/corefx/blob/475c578fc77e16344679e337276c91a70fe1c79c/src/System.Runtime.Extensions/src/System/Environment.Unix.cs#L89